docs/installationUbuntu20.04.md

```diff
@@ -0,0 +1,169 @@
+sudo apt update && sudo apt install yarn
+```
+* Rust
+```
+curl https://sh.rustup.rs -sSf | sh
+```
+* Cargo
+```
+sudo apt install cargo
+```
+
+## Update ubuntu
+```
+sudo apt-get update
+sudo apt-get upgrade
+sudo apt-get dist-upgrade
+```
+
+## Install dependencies
+```
+sudo apt-get install git ssh curl wget gcc g++ cmake \
+libstxxl-dev libxml2-dev libsparsehash-dev libbz2-dev zlib1g-dev libzip-dev libgomp1 \
+pkg-config libgdal-dev libtbb-dev psmisc build-essential postgresql postgis \
+clang libboost-all-dev libexpat1-dev libjsoncpp-dev libncurses5-dev lua5.3 liblua5.3-dev \
+powerline fonts-powerline zsh
+```
+
+## Create sources directory
+```
+cd ~
+mkdir sources
+```
+
+## Install and compile capnproto
+```
+cd sources
+curl -O https://capnproto.org/capnproto-c++-0.8.0.tar.gz
+tar zxf capnproto-c++-0.8.0.tar.gz
+cd capnproto-c++-0.8.0
+./configure
+make -j6 check
+sudo make install
+```
+
+## Generate ssh key
+```
+ssh-keygen -t rsa -b 4096 -C "YOUR_EMAIL"
+```
+* Press Enter to use the default
+* Press Enter again twice to use no passphrase
+
+```
+eval "$(ssh-agent -s)"
+```
+
+* Make sure you get "Agent pid [SOME_NUMBER]"
+```
+ssh-add -K ~/.ssh/id_rsa
+nano ~/.ssh/id_rsa.pub
+```
+* Copy this key to github (In github: User icon->Settings->SSH and GPG keys->New SSH key)
+
+## Install zsh and oh-my-zsh
+```
+git clone https://github.com/robbyrussell/oh-my-zsh.git ~/.oh-my-zsh
+cp ~/.oh-my-zsh/templates/zshrc.zsh-template ~/.zshrc
+cd ~
+nano .zshrc
+```
+* Change ZSH_THEME to "agnoster"
+Save and quit nano (ctrl-o, enter, ctrl-x)
+```
+chsh -s /bin/zsh
+cd .oh-my-zsh
+sh tools/upgrade.sh
+git clone https://github.com/zsh-users/zsh-syntax-highlighting.git "$HOME/.zsh-syntax-highlighting" --depth 1
+echo "source $HOME/.zsh-syntax-highlighting/zsh-syntax-highlighting.zsh" >> "$HOME/.zshrc"
+````
+
+## Clone repos
+```
+cd ~/sources
+git clone ssh://git@github.com/Project-OSRM/osrm-backend
+git clone ssh://git@github.com/kaligrafy/trRouting
+git clone ssh://git@github.com/chairemobilite/transition
+```
+
+## Setup Postgresql
+
+```
+sudo -u postgres psql
+```
+* Choose a password: `\password`
+* Quit psql: `\q`
+
+## Install OSRM
+
+* Build and install
+```
+cd ~/sources/osrm-backend
+git checkout tags/v5.26.0
+git switch -c 5.26.0
+mkdir -p build
+cd build
+cmake .. -DCMAKE_BUILD_TYPE=Release
+cmake --build .
+sudo cmake --build . --target install
+```
+
+## Install trRouting
+```
+cd ~/sources/trRouting
+autoreconf -i
+./configure
+make
+sudo make install
+````
+
+## Install and setup node and yarn
+```
+curl -sL https://deb.nodesource.com/setup_13.x | sudo bash -
+sudo apt-get install -y nodejs
+curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+sudo apt-get update && sudo apt-get install yarn
+```
+
+## Install Transition
+```
+cd ~/sources/transition
+sudo apt-get install build-essential
```

Comment:
A bit surprised that build-essential are not higher in the doc

Reply:
Actually, it is higher in the doc too! We can remove from here...